### PR TITLE
feat: download ovms models to requested precision only

### PR DESCRIPTION
**Pull request by vli11**
Closes: #326

## PR Checklist

- [x] Added label to the Pull Request for easier discoverability and search
- [x] Commit Message meets guidelines as indicated in the URL https://github.com/intel-retail/automated-self-checkout/blob/main/CONTRIBUTING.md
- [x] Every commit is a single defect fix and does not mix feature addition or changes
- [ ] Unit Tests have been added for new changes
- [ ] Updated Documentation as relevant to the changes
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## What are you changing?
download ovms models to requested precision only

## Issue this PR will close

close: #326 

## Test Instructions if applicable
1. delete previously downloaded models folder under configs/opencv-ovms/models/2022
2. cd download_models
3. ./downloadOVMSModels.sh

after step 3, you should observe models been downloaded and only the requested precision folder been created download each model folder

### CODE REVIEW

File: download_models/downloadOVMSModels.sh [hunk 1]
```diff
@@ -202,27 +203,36 @@ fi
...
 
 configFile="$modelDir"/config_template.json
-mapfile -t modelNames < <(docker run -i --rm -v ./:/app ghcr.io/jqlang/jq -r '.model_config_list.[].config.name' < "$configFile")
+mapfile -t model_base_path < <(docker run -i --rm -v ./:/app ghcr.io/jqlang/jq -r '.model_config_list.[].config.base_path' < "$configFile")
 
-for eachModel in "${modelNames[@]}" ; do
-    ret=$(isModelDownloaded "$eachModel")
+for eachModelBasePath in "${model_base_path[@]}" ; do
+    eachModel=$(echo "$eachModelBasePath" | rev | cut -d '/' -f 2 | rev)
+    precision=$(echo "$eachModelBasePath" | rev | cut -d '/' -f 1 | rev)
```
Comment (jim-wang-intel) [inline download_models/downloadOVMSModels.sh:227]: suggest using awk is simpler and more performant:
```suggestion
    eachModel=$(echo "$eachModelBasePath" | awk -F/ '{print $(NF-1)}')
    precision=$(echo "$eachModelBasePath" | awk -F/ '{print $NF}')
```

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Review by ejlee3 — APPROVED** [review]
LGTM

File: download_models/downloadOVMSModels.sh [hunk 2]
```diff
@@ -202,27 +209,51 @@ fi
...
+    precision=$2
     for m in "$modelDir"/* ; do
         if [ "$(basename "$m")" = "$modelName" ]
         then
-            echo "downloaded"
-            return 0
+            if [ -z "$precision" ]
+            then
+                # empty precision, but found the modelName, so assume it is downloaded
+                echo "downloaded"
```
Comment (NeethuES-intel) [inline download_models/downloadOVMSModels.sh:219]:
```suggestion
                echo "Model already downloaded"
```

File: download_models/downloadOVMSModels.sh [hunk 3]
```diff
@@ -202,27 +209,51 @@ fi
...
+                return 0
+            else
+                for precision_folder in "$modelDir"/"$modelName"/* ; do
+                    if [ "$(basename "$precision_folder")" = "$precision" ]
+                    then
+                        echo "downloaded"
```
Comment (NeethuES-intel) [inline download_models/downloadOVMSModels.sh:225]:
```suggestion
                        echo "Model already downloaded"
```

**Review by NeethuES-intel — APPROVED** [review]
Small comment on message echo. LGTM

**Pull request merged**